cmd/apiregister-gen/generators/versioned_generator.go

```diff
@@ -69,17 +69,30 @@ var (
 	ApiVersion = builders.NewApiVersion("{{.Group}}.{{.Domain}}", "{{.Version}}").WithResources(
 		{{ range $api := .Resources -}}
 		{{$api.Group}}{{$api.Kind}}Storage,
-		builders.NewApiResource( // Resource status endpoint
-			{{ $api.Group }}.Internal{{ $api.Kind }}Status,
-			{{.Kind}}SchemeFns{},
-			func() runtime.Object { return &{{ $api.Kind }}{} },     // Register versioned resource
-			func() runtime.Object { return &{{ $api.Kind }}List{} }, // Register versioned resource list
-			&{{ $api.Group }}.{{ $api.Kind }}StatusStrategy{builders.StatusStorageStrategySingleton},
-		),
+		{{if $api.REST }}
```

Comment:
I think you want to guard the one above this

> {{$api.Group}}{{$api.Kind}}Storage

The one you are guarding is just for the `/status` subresource, not the actual resource.